collect_app/src/main/java/org/odk/collect/android/formmanagement/CollectFormEntryControllerFactory.kt

```diff
@@ -26,9 +36,40 @@ class CollectFormEntryControllerFactory(
 
         return FormEntryController(FormEntryModel(formDef)).also {
             val externalDataHandlerPull = ExternalDataHandlerPull(externalDataManager)
-            it.addFunctionHandler(PullDataFunctionHandler(entitiesRepository, externalDataHandlerPull))
+            it.addFunctionHandler(
+                PullDataFunctionHandler(
+                    entitiesRepository,
+                    externalDataHandlerPull
+                )
+            )
             it.addPostProcessor(EntityFormFinalizationProcessor())
+
+            if (settings.getBoolean(ProjectKeys.KEY_DEBUG_FILTERS)) {
```

Comment:
I don't think it's convenient for the QA team or us to enable that option every time we install a new version of the app, which might happen quite often. Wouldn't it be better to always use that strategy in debug mode?

Reply:
If it defaults to on, I think we'll need to ignore the setting in non debug builds. That seems like good idea anyway though so I'll go with that.

Reply:
I've realized this actually gets a bit more complicated because QA usually test with `selfSignedRelease` (not debuggable) and we probably still want to be able to disable it if it's annoying for something like a demo. I also don't think we want it enabled in beta by default, but still want it available?

Reply:
So we can enable it by default for `selfSignedRelease`? I think that would be ok.